### PR TITLE
Add: MeshCutter wasm.

### PR DESCRIPTION
**Pull request by gonnavis**
Follow & wasm version of https://github.com/webaverse/metaverse-utils/pull/1

Related PR: https://github.com/webaverse/app/pull/2211 , https://github.com/webaverse/mesh-cut-example/pull/1

First usable version MeshCutter wasm.

Need many improvement afterwards, such as:
1. Not hard-code cut `plane`.
2. Use `struct` to make codes less verbose.
3. Do not hard code output alloc length. Try to pass back wasm output length to js.

### CODE REVIEW

**Review by avaer — APPROVED** [review]
Amazing work 🥇

**Comment by avaer** [comment]
I'm ok to merge this if we can resolve the rebuild conflict.

**Comment by avaer** [comment]
> Do not hard code output alloc length. Try to pass back wasm output length to js.
You can `malloc` in the C++, pass back the address and the length, and then `_free` that address from the JS (yes it's legal). Address and length lets you do whatever you want to the data in Javascript.

**Comment by avaer** [comment]
> Not hard-code cut plane.
> Use struct to make codes less verbose.

These would be nice to have, but do not change the functionality so we can merge already. I agree they would be nice.

**Comment by gonnavis** [comment]
OK, thanks! Let me keep improve it.

**Comment by gonnavis** [comment]
> C++ knows the size, so you can `malloc` in the C++, pass back the address and the length to JS, and then `Module._free` that address from the JS (yes it's legal). Address and length lets you do whatever you want to the data in Javascript.

Does `allocator.freeAll();` in js sufficient? https://github.com/webaverse/app/commit/c011b3e4757c1e2119d776146a5aba8e57d41d7c

**Comment by avaer** [comment]
> Does allocator.freeAll(); in js sufficient? webaverse/app@c011b3e

Yes I believe that works.

**Review by avaer — APPROVED** [review]

**Comment by avaer** [comment]
> Does allocator.freeAll(); in js sufficient?

This is only sufficient for the things allocator allocated. Obviously if you do `malloc` yourself you must also `free` yourself. That has nothing to do with `Allocator`.

File: cut.cc [hunk 1]
```diff
@@ -0,0 +1,1671 @@
...
+  std::vector<float> points1;
+  std::vector<float> points2;
+
+  std::vector<float> normals1;
+  std::vector<float> normals2;
+
+  std::vector<float> uvs1;
+  std::vector<float> uvs2;
+
+  for(int i=0;i<facesCount;i++){
```
[the remark this answers is not in the document]
Comment (avaer) [reply cut.cc:167]: I could be a comment here.

**Pull request merged**